include/deal.II/base/polynomials_vector_anisotropic.h

```diff
@@ -0,0 +1,158 @@
+ * @ingroup Polynomials
+ */
+template <int dim>
+class PolynomialsVectorAnisotropic : public TensorPolynomialsBase<dim>
+{
+public:
+  /**
+   * Constructor. Creates all basis functions for anisotropic vector-valued
+   * polynomials of given degrees in normal and tangential directions,
+   * respectively.
+   */
+  PolynomialsVectorAnisotropic(
+    const unsigned int               degree_normal,
+    const unsigned int               degree_tangential,
+    const std::vector<unsigned int> &polynomial_ordering);
+
+  /**
+   * Copy constructor.
+   */
+  PolynomialsVectorAnisotropic(const PolynomialsVectorAnisotropic &other) =
+    default;
+
+  /**
+   *  Compute the value and certain derivatives of each anisotropic polynomial
+   * at
+   * @p unit_point.
+   *
+   * The size of the vectors must either be zero or equal <tt>n()</tt>. In
+   * the first case, the function will not compute these values.
+   */
+  void
+  evaluate(const Point<dim>            &unit_point,
+           std::vector<Tensor<1, dim>> &values,
+           std::vector<Tensor<2, dim>> &grads,
+           std::vector<Tensor<3, dim>> &grad_grads,
+           std::vector<Tensor<4, dim>> &third_derivatives,
+           std::vector<Tensor<5, dim>> &fourth_derivatives) const override;
+
+  /**
+   * Return the name of the space, which is
+   * <tt>PolynomialsVectorAnisotropic</tt>.
+   */
+  std::string
+  name() const override;
+
+  /**
+   * Return the number of polynomials in the space without requiring to
+   * build an object of PolynomialsVectorAnisotropic. This is required by the
+   * FiniteElement classes.
+   */
+  static unsigned int
+  n_polynomials(const unsigned int normal_degree,
+                const unsigned int tangential_degree);
+
+  /**
+   * Return degree of polynomials in tangential direction
+   */
+  unsigned int
+  get_tangential_degree() const;
+
+  /**
+   * Return degree of polynomials in normal direction
+   */
+  unsigned int
+  get_normal_degree() const;
+
+  /**
+   * @copydoc TensorPolynomialsBase::clone()
+   */
+  virtual std::unique_ptr<TensorPolynomialsBase<dim>>
+  clone() const override;
+
+  /**
+   * Compute the generalized support points in the ordering used by the
+   * polynomial shape functions. Note that these points are not support points
+   * in the classical sense as the Lagrange polynomials of the different
+   * components have different points, which need to be combined in terms of
+   * Piola transformations.
+   */
+  std::vector<Point<dim>>
+  get_polynomial_support_points() const;
+
+private:
+  /**
+   * The given degree in the normal direction.
+   */
+  const unsigned int normal_degree;
+
+  /**
+   * The given degree in the tangential direction.
+   */
+  const unsigned int tangential_degree;
+
+  /**
+   * An object representing the polynomial space for a single component.
+   */
+  const AnisotropicPolynomials<dim> polynomial_space;
+
+  /**
+   * Renumbering from lexicographic to hierarchic order.
```

Comment:
Maybe be more specific here: 
```suggestion
   * Renumbering from lexicographic order that is used in the underlying
   * scalar anisotropic tensor-product polynomial space to the numbering
   * used by the finite element.
```